lib/iris/analysis/stats.py

```diff
@@ -4,13 +4,16 @@
 # See LICENSE in the root of the repository for full licensing details.
 """Statistical operations between cubes."""
 
+import dask.array as da
 import numpy as np
-import numpy.ma as ma
 
 import iris
-from iris.util import broadcast_to_shape
+from iris.common import SERVICES, Resolve
+from iris.common.lenient import _lenient_client
+from iris.util import _mask_array
 
 
+@_lenient_client(services=SERVICES)
```

Comment:
You'll note that leniency currently only has one option - `"maths"` - which `Resolve` responds to, e.g.:

https://github.com/SciTools/iris/blob/b3b9ddeeaffb7372481815599c6a1469b7694307/lib/iris/common/resolve.py#L1764

Do you consider `pearsonr()` to fall under the `"maths"` category? Or would the API make more sense if it came under a separate category.

Either way, something should be communicated in the docstring so that the user knows they have this level of control. Perhaps some links to [Lenient Cube Maths](https://scitools-iris.readthedocs.io/en/latest/further_topics/lenient_maths.html) or similar.

```
:ref:`lenient maths`
```

Reply:
The only other place I know of that uses `Resolve` is `mask_cube`, where we just stated that it follows [the same broadcasting approach as cube arithmetic](https://scitools-iris.readthedocs.io/en/stable/_modules/iris/util.html#mask_cube).  I would say that `pearsonr` is more mathsy than `mask_cube` is so we should maybe just add a similar note to the part of the docstring that states the two cubes must be broadcastable to each other.

On a practical note, I don't really know what would be involved in creating a separate category, or what the implications would be.  I also suspect that the vast majority of users don't think about it and leave all the leniency settings to default.

Reply:
I've looked around and no other operations mention that they respond to the leniency framework. That's either deliberate, or if not then we would need to run through the API as a whole. Either way it's not a responsibility of this PR.

Reply:
>we should maybe just add a similar note to the part of the docstring that states the two cubes must be broadcastable to each other.

Sounds good 👍

Reply:
Done